lib/src/components/stepper_input/stepper_input.dart

```diff
@@ -0,0 +1,182 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+import '../../../zeta_flutter.dart';
+
+/// A stepper input, also called numeric stepper, is a common UI element that allows uers to input a number or value simply by clicking the plus and minus buttons.
+class ZetaStepperInput extends StatefulWidget {
+  /// Creates a new [ZetaStepperInput]
+  const ZetaStepperInput({
+    this.rounded = true,
+    this.size = ZetaWidgetSize.medium,
+    this.initialValue,
+    this.min,
+    this.max,
+    this.onChange,
+    super.key,
+  }) : assert(
+          (min == null || (initialValue ?? 0) >= min) && (max == null || (initialValue ?? 0) <= max),
+          'Initial value must be inside given min and max values',
+        );
+
+  /// {@macro zeta-component-rounded}
+  final bool rounded;
+
+  /// The size of the stepper input.
+  final ZetaWidgetSize size;
+
+  /// The initial value of the stepper input.
+  ///
+  /// Must be in the bounds of [min] and [max] (if given).
+  final int? initialValue;
+
+  /// The minimum value of the stepper input.
+  final int? min;
+
+  /// The maximum value of the stepper input.
+  final int? max;
+
+  /// Called with the value of the stepper whenever it is changed.
+  final ValueChanged<int>? onChange;
+
+  @override
+  State<ZetaStepperInput> createState() => _ZetaStepperInputState();
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty<bool>('rounded', rounded))
+      ..add(EnumProperty<ZetaWidgetSize>('size', size))
+      ..add(IntProperty('initialValue', initialValue))
+      ..add(IntProperty('min', min))
+      ..add(IntProperty('max', max))
+      ..add(ObjectFlagProperty<ValueChanged<int>?>.has('onChange', onChange));
+  }
+}
+
+class _ZetaStepperInputState extends State<ZetaStepperInput> {
+  final TextEditingController _controller = TextEditingController();
+  int _value = 0;
+  late final bool _disabled;
+
+  @override
+  void initState() {
+    super.initState();
+    _disabled = widget.onChange == null;
+    if (widget.initialValue != null) {
```

Comment:
we should check if initial value violates the min / max somewhere

Reply:
I have this in an assert